pkg/httphelper/httphelper.go

```diff
@@ -6,14 +6,69 @@ import (
 	"net/http"
 )
 
-func Error(w http.ResponseWriter, err error) {
+type ErrorCode int
+
+const (
+	NotFoundError ErrorCode = iota
+	ObjectExistsError
+	SyntaxError
+	ValidationError
+	ServerError
+)
+
+var errorStrings = map[ErrorCode]string{
+	NotFoundError:     "not_found",
```

Comment:
Does it makes sense to add two types of not found?  i.e. to address this problem: https://github.com/flynn/flynn/pull/631/files#diff-e088955583d3e8787905eb3bdace3b49R88 .  Doing the host-rpc-to-http branch had a lot of head-on-desk debuging moments where I mistook a 404 for invalid route format when it was actually a misparsed ID, or vice versa.

Perhaps
```
NotFoundError
ObjectNotExistsError
```
(to parallel the `ObjectExistsError` name)